File: app/src/main/java/org/worshipsongs/domain/Topics.java
package org.worshipsongs.domain;

import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;

/**
 * Author : Madasamy
 * Version : 3.x
 */

public class Topics
{
    private int id;
    private String name;
    private String tamilName;
    private String defaultName;
    private int noOfSongs;

    public Topics()
    {
        //Do nothing
    }

    public int getId()
    {
        return id;
    }

    public void setId(int id)
    {
        this.id = id;
    }

    public String getName()
    {
        return name;
    }

    public void setName(String name)
    {
        this.name = name;
    }

    public String getTamilName()
    {
        return tamilName;
    }

    public void setTamilName(String tamilName)
    {
        this.tamilName = tamilName;
    }

    public String getDefaultName()
    {
        return defaultName;
    }

    public void setDefaultName(String defaultName)
    {
        this.defaultName = defaultName;
    }

    public Topics(String name)
    {
        this.setName(name);
    }

    public int getNoOfSongs()
    {
        return noOfSongs;
    }

    public void setNoOfSongs(int noOfSongs)
    {
        this.noOfSongs = noOfSongs;
    }

    @Override
    public String toString()
    {
        return "Topics{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", tamilName='" + tamilName + '\'' +
                ", defaultName='" + defaultName + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object object)
    {
        if (object instanceof Topics) {
            Topics otherObject = (Topics) object;
            EqualsBuilder equalsBuilder = new EqualsBuilder();
            equalsBuilder.append(getName(), otherObject.getName());
            return equalsBuilder.isEquals();
        }
        return false;
    }

    @Override
    public int hashCode()
    {
        HashCodeBuilder hashCodeBuilder = new HashCodeBuilder();
        hashCodeBuilder.append(getName());
        return hashCodeBuilder.hashCode();
    }

}

File: app/src/main/java/org/worshipsongs/domain/Topics.kt
package org.worshipsongs.domain

import org.apache.commons.lang3.builder.EqualsBuilder
import org.apache.commons.lang3.builder.HashCodeBuilder

/**
 * Author : Madasamy
 * Version : 3.x
 */

class Topics
{
    var id: Int = 0
    var name: String? = null
    var tamilName: String? = null
    var defaultName: String? = null
    var noOfSongs: Int = 0

    constructor()
    {
        //Do nothing
    }

    constructor(name: String)
    {
        this.name = name
    }

    override fun toString(): String
    {
        return "Topics{" + "id=" + id + ", name='" + name + '\''.toString() + ", tamilName='" + tamilName + '\''.toString() + ", defaultName='" + defaultName + '\''.toString() + '}'.toString()
    }

    override fun equals(`object`: Any?): Boolean
    {
        if (`object` is Topics)
        {
            val otherObject = `object` as Topics?
            val equalsBuilder = EqualsBuilder()
            equalsBuilder.append(name, otherObject!!.name)
            return equalsBuilder.isEquals
        }
        return false
    }

    override fun hashCode(): Int
    {
        val hashCodeBuilder = HashCodeBuilder()
        hashCodeBuilder.append(name)
        return hashCodeBuilder.hashCode()
    }

}
